file.go

```diff
@@ -1,37 +1,115 @@
-package jcsv
+package main
 
-import "os"
+import (
+	"encoding/csv"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	data []uint8
+	csvData [][]string
+	csvHeader []string
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	var f file
+	f.data=nil
+	fileStream,fileOpenError := os.Open(path)
+	if fileOpenError!=nil{
+		return file{},fileOpenError
+	}
+	var fileReadError error
+	f.data,fileReadError=ioutil.ReadAll(fileStream)
+	if fileReadError!=nil {
+		return file{},fileReadError
+	}
+	return f,nil
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var fileRead file
+	var fileReadError error
+	fileRead.data,fileReadError=ioutil.ReadAll(f)
+	if fileReadError!=nil {
+		return file{},fileReadError
+	}
+	return fileRead,nil
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	var f file
+	f.data=nil
+	f.csvData=nil
+	f.csvHeader=nil
+	csvFile, err := os.Open(path)
+	if err != nil {
+		fmt.Println(err)
+		return file{},err
+	}
+	readBuffer,readError := csv.NewReader(csvFile).ReadAll()
+	if readError!=nil{
+		return file{},readError
+	}
+	if hasHeaders{
+		for j:=0;j<len(readBuffer[0]);j++{
+			f.csvHeader=append(f.csvHeader,readBuffer[0][j])
+		}
+	}
+	var i int
+	if hasHeaders{
+		i=1
+	}else{
+		i=0
+	}
+	for ;i<len(readBuffer);i++{
+		f.csvData = append(f.csvData, readBuffer[i])
+	}
```

Comment:
`readBuffer` is `[][]string`, and `f.csvData` is also `[][]string`.
Can't we do something like `f.csvData=readBuffer`?

Reply:
> `readBuffer` is `[][]string`, and `f.csvData` is also `[][]string`. Can't we do something like `f.csvData=readBuffer`?

Right now, I am not sure, assignment operator does shallow copy or deep, to be on the safe side, I tried doing this through loop.

Reply:
For array, it copies its values.
For slices, it copies its references.

https://stackoverflow.com/questions/25083235/go-assign-multiple-return-value-function-to-new-and-old-variable

Reply:
There is also a built-in function in golang for the `copy()`. You can also check how it works. 
https://pkg.go.dev/builtin#copy